src/protean/core/repository/factory.py

```diff
@@ -16,12 +18,16 @@ class RepositoryFactory:
         be let go.
     """
 
+    # EntityRecord Inner Class, implemented as a namedtuple for ease of use.
+    #   This class will store attributes related to Entity and Models, and will be objects
+    #   in the registry dictionary.
+    EntityRecord = namedtuple(
```

Comment:
We should use the new dataclass here, which will avoid the need for using `_replace` for change attributes. However in 3.6 we will need to install a library to support dataclass

Reply:
That's good input! 👍 I read through about `dataclasses` and do agree we could use it. However, like you said, supporting for both 3.6 and 3.7 would mean a version check somewhere in the code. Can we stay with `namedtuple` for now, because it satisfies the requirement? I chose `namedtuple` because of its immutability, and that `_replace` is the only thing weird in there. IN the future, we may find a better mechanism to avoid the model refresh altogether.